ArchiSteamFarm.OfficialPlugins.Monitoring/MonitoringPlugin.cs

```diff
@@ -0,0 +1,239 @@
+// ----------------------------------------------------------------------------------------------
+// |
+// Copyright 2015-2024 Łukasz "JustArchi" Domeradzki
+// Contact: JustArchi@JustArchi.net
+// |
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// |
+// http://www.apache.org/licenses/LICENSE-2.0
+// |
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.DataAnnotations;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Metrics;
+using System.Globalization;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using ArchiSteamFarm.Core;
+using ArchiSteamFarm.IPC.Integration;
+using ArchiSteamFarm.OfficialPlugins.Monitoring.Localization;
+using ArchiSteamFarm.Plugins;
+using ArchiSteamFarm.Plugins.Interfaces;
+using ArchiSteamFarm.Steam;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Metrics;
+using SteamKit2;
+
+namespace ArchiSteamFarm.OfficialPlugins.Monitoring;
+
+[Export(typeof(IPlugin))]
+[SuppressMessage("ReSharper", "MemberCanBeFileLocal")]
+internal sealed class MonitoringPlugin : OfficialPlugin, IWebServiceProvider, IGitHubPluginUpdates, IASF, IDisposable {
+	[JsonInclude]
+	[Required]
+	public override string Name => nameof(MonitoringPlugin);
+
+	[JsonInclude]
+	[Required]
+	public override Version Version => typeof(MonitoringPlugin).Assembly.GetName().Version ?? throw new InvalidOperationException(nameof(Version));
+
+	private bool Enabled => (ASF.GlobalConfig?.IPC ?? false) && (Config?.Enabled ?? MonitoringConfig.DefaultEnabled);
+
+	private const string MeterName = nameof(ArchiSteamFarm);
+
+	private const string MetricNamePrefix = "asf";
+
+	private Meter? Meter;
+
+	public void OnConfiguringServices(IServiceCollection services) {
+		ArgumentNullException.ThrowIfNull(services);
+
+		if (!Enabled) {
+			return;
+		}
+
+		services.AddOpenTelemetry().WithMetrics(
+			static builder => {
+				builder.AddPrometheusExporter(static config => config.ScrapeEndpointPath = "/Api/metrics");
+				builder.AddMeter("Microsoft.AspNetCore.Hosting");
+				builder.AddMeter("Microsoft.AspNetCore.Server.Kestrel");
+				builder.AddMeter(MeterName);
+
+				builder.AddView(
+					"http.server.request.duration",
+					new ExplicitBucketHistogramConfiguration {
+						Boundaries = [0, 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10]
+					}
+				);
```

Comment:
Do we really need this configuration considering that OpenTelemetry automatically uses these buckets? https://github.com/open-telemetry/opentelemetry-dotnet/blob/dd4e212d593e0ad02decc1bf22d972fa1a275618/src/OpenTelemetry/Metrics/Metric.cs#L20-L23

Reply:
No. Parts of this code stem from a test I did before `OpenTelemetry.Extensions.Hosting` had a stable release. I'll remove this.